collections/go.mod

```diff
@@ -0,0 +1,53 @@
+module cosmossdk.io/collections
```

Comment:
A few more things needs to be done when adding a new module:

- Can you add it to dependabot
- Can you add it in test.yml (under submodules)
- Can you create a `sonar-project.properties` file (like in math)

Reply:
lets do this in a follow up pr. Ideally we can scope PRs better to tooling vs logic

Reply:
Make sense, expect for the addition in CI. This should be done in the same PR as now our CI does not test it.